src/app/vault/ciphers.component.ts

```diff
@@ -80,6 +86,96 @@ export class CiphersComponent extends BaseCiphersComponent implements OnDestroy
         this.didScroll = this.pagedCiphers.length > this.pageSize;
     }
 
+    async loadOrganizationNames(): Promise<Map<string, string>> {
+        const organizations = await this.userService.getAllOrganizations();
+        const organizationNames = new Map<string, string>();
+
+        if (organizations.length > 0) {
+            organizations.forEach(organization => {
+                organizationNames.set(organization.id, organization.name);
+            });
+        }
+
+        return organizationNames;
+    }
+
+    getOrganizationById(c: CipherView): string {
+        if (c.organizationId != null) {
+            return this.organizationNames.get(c.organizationId);
+        }
+        return this.i18nService.t('myVault');
+    }
+
+    sortCiphers(sortBy: string): void {
+        if (this.sortByOptions.includes(sortBy)) {
+            if (sortBy === this.sortBy) {
+                this.sortedDescending = !this.sortedDescending;
+            } else {
+                this.sortBy = sortBy;
+                this.sortedDescending = true;
+            }
+
+            const self = this;
+
+            if (sortBy === 'lastEdited') {
+                this.ciphers.sort((a: CipherView, b: CipherView) => {
+                    if (self.sortedDescending) {
```

Comment:
I'd recommend moving this `if` condition out of the `sort` callback so that we evaluate it once only. Here it's being checked repeatedly as the list is being sorted. Same for the other sorting logic below.